internal/tools/sign1util/main.go

```diff
@@ -0,0 +1,371 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/cosesign1"
+	didx509resolver "github.com/Microsoft/hcsshim/internal/did-x509-resolver"
+	"github.com/urfave/cli"
+)
+
+func checkCoseSign1(inputFilename string, chainFilename string, didString string, verbose bool) (*cosesign1.UnpackedCoseSign1, error) {
+	coseBlob, err := os.ReadFile(inputFilename)
+	if err != nil {
+		return nil, err
+	}
+
+	var chainPEM []byte
+	var chainPEMString string
+	if chainFilename != "" {
+		chainPEM, err = os.ReadFile(chainFilename)
+		if err != nil {
+			return nil, err
+		}
+		chainPEMString = string(chainPEM[:])
+	}
+
+	unpacked, err := cosesign1.UnpackAndValidateCOSE1CertChain(coseBlob)
+	if err != nil {
+		fmt.Fprintf(os.Stdout, "checkCoseSign1 failed - %s\n", err)
+		return nil, err
+	}
+
+	fmt.Fprint(os.Stdout, "checkCoseSign1 passed\n")
+	if verbose {
+		fmt.Fprintf(os.Stdout, "iss: %s\n", unpacked.Issuer)
+		fmt.Fprintf(os.Stdout, "feed: %s\n", unpacked.Feed)
+		fmt.Fprintf(os.Stdout, "cty: %s\n", unpacked.ContentType)
+		fmt.Fprintf(os.Stdout, "pubkey: %s\n", unpacked.Pubkey)
+		fmt.Fprintf(os.Stdout, "pubcert: %s\n", unpacked.Pubcert)
+		fmt.Fprintf(os.Stdout, "payload:\n%s\n", string(unpacked.Payload[:]))
+	}
+	if len(didString) > 0 {
+		if len(chainPEMString) == 0 {
+			chainPEMString = unpacked.ChainPem
+		}
+		didDoc, err := didx509resolver.Resolve(chainPEMString, didString, true)
+		if err == nil {
+			fmt.Fprintf(os.Stdout, "DID resolvers passed:\n%s\n", didDoc)
+		} else {
+			fmt.Fprintf(os.Stdout, "DID resolvers failed: err: %s doc:\n%s\n", err.Error(), didDoc)
+		}
+	}
+	return unpacked, err
+}
+
+var createCmd = cli.Command{
+	Name:  "create",
+	Usage: "",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "claims",
+			Usage: "filename of payload",
+			Value: "fragment.rego",
+		},
+		cli.StringFlag{
+			Name:  "content-type",
+			Usage: "payload content type",
+			Value: "application/unknown+json",
+		},
+		cli.StringFlag{
+			Name:  "chain",
+			Usage: "key or cert file to use (pem)",
+			Value: "chain.pem",
+		},
+		cli.StringFlag{
+			Name:  "key",
+			Usage: "key to sign with - private key of the leaf of the chain",
+			Value: "key.pem",
+		},
+		cli.StringFlag{
+			Name:     "algo",
+			Usage:    "PS256, PS384 etc (required)",
+			Required: true,
+		},
+		cli.StringFlag{
+			Name:  "out",
+			Usage: "output file (default: out.cose)",
+			Value: "out.cose",
+		},
+		cli.StringFlag{
+			Name:  "salt",
+			Usage: "salt type [rand|zero] (default: rand)",
```

Comment:
Nit: we should probably add error checking for the `salt` and `policy` flags in a future PR, to make sure the values provided are valid.